godot-core/src/obj/raw.rs

```diff
@@ -460,14 +464,9 @@ where
         Self::from_obj_sys_weak(obj)
     }
 
-    fn new_with_init(init: impl FnOnce(&mut Self)) -> Self {
-        let mut obj = Self {
-            obj: std::ptr::null_mut(),
-            cached_rtti: None,
-        };
-
-        init(&mut obj);
-        obj
+    unsafe fn new_with_init(init: impl FnOnce(sys::GDExtensionTypePtr)) -> Self {
+        // `new_with_uninit` creates an initialized pointer.
```

Comment:
This comment is not quite clear to me -- it's the `init()` call below and not `new_with_uninit()` that creates the initialized pointer, no?

Reply:
`new_with_uninit` creates an initialized null-object and then passes a pointer to it's `obj` field to the init function. so the pointer it passes to the init function is initialized. thus it is fine to use `new_with_uninit` as the implementation for `new_with_init` since the pointer is initialized. unlike many other implementations which pass a pointer pointing to uninitialized data.